File: ランダムソート/rand_sort.ts
const randomSort = <T>(arr: T[]): T[] => {
  const randint = (min: number, max: number): number => Math.floor((max - min + 1) * Math.random())
  for (let i = arr.length - 1; i > 0; i--) {
    const j = randint(0, i)
    const tmp = arr[i]
    arr[i] = arr[j]
    arr[j] = tmp
  }
  return arr
}

File: ランダムソート/rand_sort.py
from random import randint
from typing import TypeVar


T = TypeVar("T")


def RandomSort(arr: list[T]) -> list[T]:
    """Fisher-Yatesアルゴリズムを用いてランダムソートを行う"""

    for i in reversed(range(1, len(arr))):
        j = randint(0, i)
        arr[i], arr[j] = arr[j], arr[i]
    return arr
